pkg/broker/broker.go

```diff
@@ -441,8 +523,19 @@ func (broker *PksServiceBroker) servicesReady(instanceID string) (bool, error) {
 }
 
 func (broker *PksServiceBroker) podsReady(instanceID string) (string, bool, error) {
-	cluster, err := broker.clusterFactory.DefaultCluster()
-	if err != nil {
+	var cluster k8s.Cluster
```

Comment:
The `podsReady` helper should't re-lookup the cluster: the caller already has done that work.